### PR TITLE
3.8: chop request functions into handy bits, reuse code (#14001)

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/14001

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/15021/ testrun blue.

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,12 @@
 v3.8.0 (XXXX-XX-XX)
 -------------------
 
+ arangosh client behavior:
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* Change arangosh client behavior:
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+  - *_RAW methods will never add a `body` to HEAD responses
+  - *_RAW methods will now always return velocypack-typed responses in Buffers
+  - `--server.force-json` will now be applied as default, overrideable
+   by user code
```
Comment (jsteemann) [inline CHANGELOG:8]:
```suggestion
    by user code
```

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by dothebart** [comment]
testrun succeeded except for mac run because of test host configuration issues.

**Pull request merged**